src/x86_64/unwind_rule.rs

```diff
@@ -19,6 +20,120 @@ pub enum UnwindRuleX86_64 {
     },
     /// (sp, bp) = (bp + 16, *bp)
     UseFramePointer,
+    /// sp = sp + 8 * (offset + register count)
+    /// This supports the common case of pushed callee-saved registers followed by a stack
+    /// allocation. Up to 8 registers can be stored, which covers all callee-saved registers (aside
+    /// from RSP which is implicit).
+    ///
+    /// The registers are stored in a separate compressed ordering to facilitate restoring register
+    /// values if desired. If not for this we could simply store the total offset.
+    OffsetSpAndPopRegisters(UnwindRuleOffsetSpAndPopRegisters),
```

Comment:
What is the reason to record the full set of register into an unwind rule? AFAIU, the goal of framehop is to only get a stack trace, not recover all registers in the stack frame(s). I think this should just use OffsetSp, without extra bookkeeping for all the other registers. From my testing with a variety of binaries built with MSVC, MinGW or Clang, using the same strategy as Unix of tracking RIP, RBP and RSP only is sufficient to unwind all their frames.

Reply:
This is a bit of a deviation from framehop's initial mission statement. Alex wants the other registers for crash unwinding, and I relented and said "if I don't notice any perf regression from it, it's probably ok."

Reply:
It's tricky since if you wanted framehop to stay true to that mission, a library which unwound _and_ restored registers in a very efficient manner could look almost identical, so it makes more sense to add to framehop. Admittedly you could also fork+rename+re-mission :P